packages/bpk-component-scrollable-calendar/src/BpkScrollableCalendarGridList.js

```diff
@@ -49,87 +52,121 @@ class BpkScrollableCalendarGridList extends React.Component {
     );
     const months = getMonthsArray(startDate, monthsCount);
 
-    const cache = new CellMeasurerCache({
-      fixedWidth: true,
-      defaultHeight: 276, // most common height (in px) of BpkScrollableCalendarGrid
+    // Here we calculate the height of each calendar grid item in pixels, as the `react-window` API
+    // requires that these are provided so that they can be efficiently rendered.
+    const monthItemHeights = months.map((month) => {
+      const firstDayOffset = (month.getDay() + 7 - props.weekStartsOn) % 7;
+      const monthLength = DateUtils.daysInMonth(
+        month.getYear(),
+        month.getMonth(),
+      );
+      const calendarGridSpaces = firstDayOffset + monthLength;
+      const rowCount = Math.ceil(calendarGridSpaces / COLUMN_COUNT);
+      return BASE_MONTH_ITEM_HEIGHT + ROW_HEIGHT * rowCount;
     });
+
     this.state = {
       months,
-      cache,
+      monthItemHeights,
+      outerHeight: ESTIMATED_MONTH_ITEM_HEIGHT,
     };
   }
 
+  componentDidMount = () => {
+    this.setComponentHeight();
+    if (typeof document !== 'undefined') {
+      document.addEventListener('resize', this.setComponentHeight);
+      document.addEventListener('orientationchange', this.setComponentHeight);
+      document.addEventListener('fullscreenchange', this.setComponentHeight);
+    }
+  };
+
+  componentWillUnmount = () => {
+    if (typeof document !== 'undefined') {
+      document.removeEventListener('resize', this.setComponentHeight);
```

Comment:
same as above -  event listener for resize will actually need to be attached on the `window`, not the `document`